packages/magento-customer/components/AccountSignInUpForm/AccountSignInUpForm.tsx

```diff
@@ -33,9 +34,18 @@ export function AccountSignInUpForm(props: AccountSignInUpFormProps) {
 
   const { email, firstname = '' } = customerQuery.data?.customer ?? {}
   const { mode, form, autoSubmitting, submit } = useFormIsEmailAvailable({ email })
-  const { formState, muiRegister, required, watch, error } = form
+  const { formState, muiRegister, required, watch, error, setValue, trigger } = form
   const disableFields = formState.isSubmitting && !autoSubmitting
 
+  useEffect(() => {
+    const emailFromParams = router.query.email as string
+    if (!email && emailFromParams) {
+      setValue('email', emailFromParams)
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      trigger('email')
+    }
+  }, [email, setValue, trigger])
```

Comment:
Might be better to move emailFromParams out of the useEffect, you probably have a warning on the dependency array right now.